### DISCUSSION TITLE
Vorsatzworte usw. in Nachname oder Vornamen übertragen?

### DISCUSSION THREAD

**Discussion by mge-medisoftware** [Digitaler-Impfnachweis/certification-apis#143] [Q&A]
Zum Zertifikat soll laut Dokumentation Nachname und Vorname im JSON übertragen werden.

Sollen eigentlich Vorsatzworte wie "von" beim Nachnamen übertragen werden?
Bzw. Namenszusätze wie "Graf" beim Vornamen/Nachnamen übertragen werden?
Bzw. Titel wie "Dr." beim Vornamen/Nachnamen übertragen werden?

Vermutlich nicht, machen wir z.Zt. auch nicht. Frage von Praxen bzw. Patienten...(weil ja im Personalausweis evtl. so drin)

**Comment by thinkberg** [comment]
Wichtig ist, dass alle Namensinformationen enthalten sind, die im Ausweis auch in der maschinenlesbaren Zone auftauchen sollten. Das Backend nimmt eine Transliteration automatisiert aus dem Input vor.

**Reply by mge-medisoftware** [reply]
Ok, danke, soweit die Theorie. Es werden ja meist die Namensangaben von der Gesundheitskarte verwendet, die Praxen kontrollieren eher nicht den Personalausweis. 
Auch deshalb kann es also Differenzen geben..

**Reply by thinkberg** [reply]
Dies koennte zu Problemen fuehren, da der Name wie auf dem PA erwartet wird, wenn dies an der Grenze geprueft wird.

**Reply by dirkexe** [reply]
Der PA (Personalausweis) hat 3 Felder in diesem Kontext: Familienname, Vornamen und Doktorgrad (Quelle: [§5 PAuswG](https://www.gesetze-im-internet.de/pauswg/__5.html)). Die API hier verlangt, die 3 Felder des PA auf 2 Felder abzubilden. Eine eindeutige Handlungsanweisung ("Mapping"), wie mit dem Doktorgrad (Titel) zu verfahren ist, wäre zielführend für einheitliche Zertifikate. Oder steht das bereits irgendwo und ich habe es überlesen?

**Reply by mge-medisoftware** [reply]
Weiß nicht ob man den schwarzen Peter an die Praxen durchreichen sollte für die 2€ Honorar... und bei den meisten Kontrollen wird ja noch nicht mal die CovPassCheck-App verwendet (am Flughafen hoffentlich schon).
Feldlängen von 64 Zeichen sind im Extremfall ohnehin zu kurz.

"In den Datenfeldern „Name“ (Familienname und Geburtsname) sowie „Vornamen“ sind alle Namensbestandteile komplett darzustellen, soweit dies technisch entsprechend der nachstehenden Tabelle umsetzbar ist."

https://www.gesetze-im-internet.de/pauswv/anhang_3.html

**Reply by molk-ibm** [reply]
Die genannte Referenz auf den **Anhang 3 der PAuswV** führt aus, dass die Doktorgrade (und nur die) im Datenfeld `Name` aufzunehmen sind:

> Sofern neben dem Familiennamen auch ein Geburtsname vorhanden ist, ist der Geburtsname in einer eigenen Zeile einzutragen. Bei vorläufigen Personalausweisen und Ersatz-Personalausweisen ist dem Geburtsnamen die Zeichenfolge „GEB.“ unter Hinzufügung eines Leerzeichens voranzustellen.
> 
> Soweit ein oder mehrere Doktorgrade vorhanden sind, wird dieser oder werden diese im Datenfeld „Name“ eingetragen. Entsprechend der für die Eintragung des Doktorgrades bzw. der Doktorgrade benötigten Zeichenzahl verringert sich die Anzahl der verbleibenden Zeichen für den Namenseintrag.
> 

Die Feldlängen sind hier auch vorgegeben und passen leider nicht zu dem was die EU in der DCC-Spec vorgegeben hat:  je 80 Zeichen für Name und Vorname gegenüber max. 120 Zeichen in _Schriftart 2_ für das Datenfeld `Name`.
Bei der Erstellung des DCC wird der Name ggF. nach 80 Zeichen abgeschnitten.
Wie häufig das vorkommt ist allerdings unbekannt.

Was die Quelle der Daten betrifft, ist der sicherste Weg, den Personalausweis vorlegen zu lassen - in der Apotheke sollte das Pflicht sein und ja, das bedeutet, den _schwarzen Peter_ dem Aussteller _durchzureichen_.

**Reply by marcusthierfelder** [reply]
siehe hier: https://github.com/Digitaler-Impfnachweis/certification-apis/discussions/201

@thinkberg Konkretes Beispiel:
Vorname: Horst
Nachname: Fuchs
Titel Dr.

=> muss 
a) gn = "Dr. Horst" und fn = "Fuchs"
oder 
b) gn = "Horst" und fn = "Fuchs"

**Reply by molk-ibm** [reply]
Wenn das erstellte DCC gegen den Personalausweis gehalten wird bei einer Kontrolle, so sollte dieser auch für die Erfassung der Feldern Name und Vorname herangezogen werden.

Die eGK / das PVS trennt die Daten ggF. wir im o.g. Falle.
Das PAuswG tut dies nicht: es gibt nur `Name` und `Vorname`und der akademische Grad darf auf Wunsch mit in das Namensfeld aufgenommen werden.

D.h.: prüfen, ob im Personalausweis `Dr. Fuchs` oder `Fuchs` steht und davon hängt es dann ab, wie bei der Ausstellung die Felder zu belegen sind.

Das obige Besipiel müsste dann mit akademischen Grad im Namen `fn = "Dr. Fuchs", gn = "Horst"` lauten.
 (`fn` = _family name_ und `gn`= _given name_)

**Comment by Bergreiter** [comment]
In den maschinenlesbaren Namen darf der Titel laut Personalausweis nicht enthalten sein, also wäre es nicht besser den Titel dann ganz weg zulassen oder wie kann man den maschinenlesbaren Teil auf dem Zertifikat beeinflussen.

**Reply by marcusthierfelder** [reply]
Das war auch immer unser Knackpunkt. Der Titel steht nicht im maschinenlesbaren Teil und sollte nicht genutzt werden. Aber scheinbar will IBM es haben. Endlich eine konkrete Aussage.

**Reply by thomasaugsten** [reply]
Ja die Aussagen sind widersprüchlich. 
Es soll dem Ausweis folgen, aber es wird von IBM der maschinenlesbare Bereich falsch mit Titel erstellt. 
Entweder IBM filtert den Titel im maschinenlesbare Namen raus oder ermöglicht den maschinenlesbaren Namen zu setzen.

**Reply by thinkberg** [reply]
Ich wuerde mich freuen, wenn wir hier mit weniger bashing arbeiten. Die Ausstellende Stelle bestimmt die Darstellung. Dabei wird vom Backend nicht festgelegt, was dort erscheint, sondern, das was im Namen angegeben wird. Die Darstellung im Freitext des PDF ist hier im Grunde eine Bequemlichkeit. Entscheidend ist, dass Namen und Vorname so eingegeben werden, dass diese so kodiert werden koennen, dass sie in der Maschinenlesbaren Zone korrekt sind.

Es gibt im Moment zwei Formen der Eingabe a) Manuell mit Ausweis und b) automatisiert mit der eGK. Bei beiden werden die Namen voellig unterschiedlich behandelt. Die Probleme treten im Grunde nur bei der automatisierten Eingabe per eGK auf, da dort der Titel mit in eins der Namensfelder exportiert wird.

**Reply by Bergreiter** [reply]
Es soll kein Bashing sein aber eine einheitliche Antwort wäre gut
Laut der jetzigen Aussage wäre dann im Beispiel
Vorname: Horst
Nachname: Fuchs
Titel Dr.

gn = "Horst" und fn = "Fuchs"
einzutragen. 

Ist das die finale Antwort?

**Reply by molk-ibm** [reply]
> In den maschinenlesbaren Namen darf der Titel laut Personalausweis nicht enthalten sein

Wo genau steht das?

Davon hängt die _finale Antwort_ ab.

**Reply by marcusthierfelder** [reply]
Der Titel ist nicht im maschinenlesbaren Teil (Rückseite unten) des Personalausweises dabei.

**Reply by Bergreiter** [reply]
@molk-ibm  Im Personalausweisgesetz (PAuswG) §5 oder auf welche Grundlage stützt sich ihre Aussage oder die automatische Erstellung des maschinenlesbaren Namen?

**Reply by marcusthierfelder** [reply]
Ich habe meinen Perso angeschaut und Kollegen gefragt mit immer dem gleichen Ergebnis: Vorderseite unter Name steht der Titel, Rückseite unten steht der Titel nicht.
